feat(DeviceSettings): info alert with a link to the documentation

MacDada opened the pull request:
![Screenshot 2024-05-14 at 10 26 29](https://github.com/nurikk/zigbee2mqtt-frontend/assets/136261/c23036d0-ad6d-4760-9920-b1a16cdc6f02)

The link to the docs allows the user to quickly find what are the settings options for.

I personally was looking for what is "Disabled" option actually doing.

I wasn't sure how to style it -> if there is a better (maybe some kind of "standard") way of injecting this kind of info alerts, I will happily improve the PR :-)

MacDada committed:
feat(DeviceSettings): info alert with a link to the documentation

Allows the user to quickly find what are the settings options for.

I personally was looking for what is "Disabled" option actually doing.

The info icon generated using:
* https://getbootstrap.com/docs/5.0/components/alerts/#icons
* https://icons.getbootstrap.com/icons/info-circle-fill/

MacDada committed:
fix(DeviceSettings): "noreferrer" link

MacDada committed:
fix(DeviceSettings): fixed code style

prettier --check -w "./**/*.{js,jsx,tsx}"

MacDada commented on src/components/device-page/DeviceSettings.tsx:
```diff
@@ -54,14 +54,40 @@ export class DeviceSettings extends Component<DeviceSettingsProps, DeviceSetting
-                uiSchema={uiSchema}
-                fields={{ TitleField, DescriptionField }}
-            />
+            <div>
```
I will humbly listen to suggestions on how to refactor this to fix https://codebeat.co/projects/github-com-nurikk-zigbee2mqtt-frontend-dev/pull_requests/734366

Anything similar in the source code, that I can look at for inspiration? I'm pretty new to React ;)

S33G replied:
There's already libraries to handle icons
e.g.

```
<i class="fa-solid fa-circle-info"></i>
```
https://fontawesome.com/icons/circle-info?f=classic&s=solid

S33G replied:
https://www.npmjs.com/package/@fortawesome/fontawesome-free

MacDada replied:
https://github.com/nurikk/zigbee2mqtt-frontend/pull/2020/commits/2e2272f9ac9d7894f2311e8ee6867af510b8caa5

MacDada replied:
Unfortunately, the function is still "too long": https://codebeat.co/projects/github-com-nurikk-zigbee2mqtt-frontend-dev/pull_requests/734366

MacDada replied:
I could just get rid of some whitespace, but I'd rather go into a more proper direction, like extracting something like `<InfoAlert message="blahblah" />`

MacDada replied:
https://github.com/nurikk/zigbee2mqtt-frontend/pull/2020/commits/06b2fdae70352b05fda115a3627abc685d92ce0c

MacDada committed:
feat(DeviceSettings): styling of docs info alert

Centering vertically (perfectionist hell :D)

MacDada committed:
fix(DeviceSettings): docs info alert icon

https://github.com/nurikk/zigbee2mqtt-frontend/pull/2020/commits/
abc0b6e7a3e7413805d102d3ffc7768051e2a534#r1599289271

MacDada committed:
style(DeviceSettings): removed unnecessary <div>

MacDada committed:
refactor(DeviceSettings): extracted ReadTheDocsInfo

fixes https://codebeat.co/projects/github-com-nurikk-zigbee2mqtt-frontend-dev/pull_requests/734366

MacDada committed:
refactor(DeviceSettings): extracted ReadTheDocsInfo.tsx

fixes https://github.com/nurikk/zigbee2mqtt-frontend/actions/runs/9082086095/job/24957557514?pr=2020

MacDada committed:
style(DeviceSettings): whitespace

`prettier --check "./**/*.{js,jsx,tsx}" -w`

fixes https://github.com/nurikk/zigbee2mqtt-frontend/actions/runs/9082225515/job/24958031689

I actually think it's more readable with the code being split into multiply lines…
Not sure why the `prettier` did not like it ;p

nurikk-sa commented on src/components/ReadTheDocsInfo.tsx:
```diff
@@ -0,0 +1,16 @@
+                <i className="fa-solid fa-circle-info fa-2xl me-2"></i>
+
+                <a href={docsUrl} target="_blank" rel="noreferrer" className="alert-link align-middle">
+                    Read about this in the documentation…
```
this calls for translation

MacDada replied:
https://github.com/nurikk/zigbee2mqtt-frontend/pull/2020/commits/0eb757b162caf99b85d638a178f9af53852c9670

MacDada committed:
fix(ReadTheDocsInfo): i18n

fixes https://github.com/nurikk/zigbee2mqtt-frontend/pull/2020/commits/
5e2778062d202eb99a16aa4c191977a151fb561a#r1604031584

nurikk commented:
Terrific job! Thanks
